### ISSUE TITLE
AutoSlash not working when prefix is specified in mountService

### ISSUE THREAD

**Issue by amarrella** [http4s/http4s#1378]
I tested this in http4s 0.18.0-M1

How to reproduce:
```
package com.example.http4squickstart

import io.circe._
import org.http4s._
import org.http4s.circe._
import org.http4s.server._
import org.http4s.dsl._
import cats.effect.IO

object HelloWorld {
  val service = HttpService[IO] {
    case GET -> Root / "hello" / name =>
      Ok(Json.obj("message" -> Json.fromString(s"Hello, ${name}")))
  }
}
```

```
package com.example.http4squickstart

import cats.effect.IO
import fs2.Stream

import scala.util.Properties.envOrNone
import org.http4s.util.StreamApp
import org.http4s.server.blaze.BlazeBuilder
import org.http4s.server.middleware.AutoSlash

object BlazeExample extends StreamApp[IO] {

  val port : Int              = envOrNone("HTTP_PORT") map (_.toInt) getOrElse 8080
  val ip   : String           = "0.0.0.0"

  override def stream(args: List[String], requestShutdown: IO[Unit]): Stream[IO, Nothing] =
    BlazeBuilder[IO]
      .bindHttp(port, ip)
      .mountService(AutoSlash(HelloWorld.service), "/test/")
      .serve

}
```

`curl -v http://localhost:8080/test/hello/alessandro` works.
```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /test/hello/alessandro HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Fri, 08 Sep 2017 17:01:52 GMT
< Content-Length: 31
< 
* Curl_http_done: called premature == 0
* Connection #0 to host localhost left intact
{"message":"Hello, alessandro"}
```

`curl -v http://localhost:8080/test/hello/alessandro/` doesn't.
```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /test/hello/alessandro/ HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Date: Fri, 08 Sep 2017 17:02:33 GMT
< Content-Length: 0
< 
* Curl_http_done: called premature == 0
* Connection #0 to host localhost left intact
```

If I remove the prefix and curl without the prefix AutoSlash works.

**Comment by amarrella** [comment]
Amazing, thank you!

**Comment by rossabaker** [comment]
This will be available in 0.18.0-SNAPSHOT now, and we'll look to do an -M2 release real soon.

**Comment by OneWebPro** [comment]
I found same issue in version 1.0.0-SNAPSHOT and 0.18.14. Can someone take a look? 

```
import cats.effect._
import org.http4s._
import org.http4s.dsl.io._
import org.http4s.server.blaze.BlazeBuilder
import fs2.StreamApp.ExitCode
import fs2.{Stream, StreamApp}
import org.http4s.server.middleware.AutoSlash

import scala.concurrent.ExecutionContext.Implicits.global

object Main extends StreamApp[IO] {

  val helloWorldService = HttpService[IO] {
    case GET -> Root / "test" => Ok("test")
  }

  override def stream(args: List[String], requestShutdown: IO[Unit]): Stream[IO, ExitCode] =
    BlazeBuilder[IO]
      .bindHttp(8080, "localhost")
      .mountService(AutoSlash(helloWorldService), "/public/")
      .serve
}
```

`GET http://localhost:8080/public/test/` returns NotFound response where `GET http://localhost:8080/public/test` returns `test` as expected.

**Comment by rossabaker** [comment]
Fixed by #1938
